cil_download.rb

```diff
@@ -15,28 +15,44 @@
 results = IO.readlines(ARGV[2])
 
 results.each do |json_record|
-  json_record = json_record.strip
-  identifier = json_record.split('/').last.split('.')[0]
-
-  FileUtils.copy_file(json_record, "#{data_path}/#{identifier}.json")
-  file = File.read(json_record)
-  metadata = JSON.parse(file)
-
-  video_format = metadata['CIL_CCDB']['Data_type']['Video']
-  content_files = metadata['CIL_CCDB']['CIL']['Image_files']
-  if content_files
-    content_files.each do |file|
-      file_path = file['File_path']
-      content_base_url = video_format ? video_base_url : image_base_url
-      `curl -s -o #{content_file_path}/#{file_path} #{content_base_url}/#{identifier}/#{file_path}`
+  begin
+    json_record = json_record.strip
+    identifier = json_record.split('/').last.split('.')[0]
+
+    json_file = "#{data_path}/#{identifier}.json"
+    if !File.exists?(json_file)
+      FileUtils.copy_file(json_record, json_file)
     end
-  end
 
-  alt_images = metadata['CIL_CCDB']['CIL']['Alternative_image_files']
-  if alt_images
-    alt_images.each do |file|
-        fileName = file['URL_postfix'].split('/').last
-        `curl -s -o #{content_file_path}/#{fileName} #{content_url_base}/#{file['URL_postfix']}`
+    file = File.read(json_record)
+    metadata = JSON.parse(file)
+
+    video_format = metadata['CIL_CCDB']['Data_type']['Video']
+    content_files = metadata['CIL_CCDB']['CIL']['Image_files']
+    if content_files
+      content_files.each do |file|
+        file_path = file['File_path']
+
+        content_file = "#{content_file_path}/#{file_path}"
+        if !File.exists?(content_file)
```

Comment:
same thing here, potentially we could save the if branch wrapping by going to the next file immediately if the file does exist already?

Reply:
See my comment above.

Reply:
I'm not sure this is the same issue as above. In this case, if we have the file, we don't want to download it again and can safely skip to the next file?. There's nothing 'deeper' as far as logic goes, right?

Reply:
Yep, you are right  that the logic seems a little deep here. But we need to loop through all those content files, then calculate the file path for each file and determine whether we need to download it or not.
It seems like i could be the case that there an error that prevents it from downloading all the content files at the initial round, or some files are missing and we need to rerun the script again later. So I think it's better to keep the logic to check for the existence of each file.

Reply:
Ok, I think we're not talking about exactly the same thing here, but it's not that big of a deal, it's just a syntax preference for guard clauses that I have. I don't want to hold up your work for that, because this is a great and important improvement to the script. let's merge

Reply:
Got it. Thanks. Just let you know that I did append the `if` condition to the end of the line but changed it to the `if` block. Lark's 80 characters line limitation make me hesitate for a  longer line.